python/ray/train/base_trainer.py

```diff
@@ -767,13 +767,16 @@ def _reconcile_scaling_config_with_trial_resources(
                     scaling_config = trainer_cls._validate_scaling_config(
                         scaling_config
                     )
+                else:
+                    return ScalingConfig()
+
                 scaling_config_from_trial_resources = (
                     ScalingConfig.from_placement_group_factory(trial_resources)
                 )
 
                 # This check should always pass if ResourceChangingScheduler is not
                 # used.
-                if scaling_config_from_trial_resources != scaling_config:
+                if trial_resources != scaling_config.as_placement_group_factory():
```

Comment:
Bypass the `ResourceChangingScheduler` Logic for `num_worker=1` case. We are unable to rebuild the original ScalingConfig when we have 1 worker. And we don't want to always go through this `ResourceChangingScheduler` workaround which will fail the tests.